### ISSUE TITLE
IMBObjectViewController should have its own delegate

### ISSUE THREAD

**Issue by gloubibou** [karelia/iMedia#26]
Currently [IMBObjectViewController delegate] calls through to self.libraryController.delegate

For one, this seems an odd thing to do. The library controller's delegate does not necessarily conform to the IMBObjectViewControllerDelegate protocol.

The current implementation is actually causing trouble for the implementation of badges: Requests for badges are directed to the [IMBObjectViewController delegate] . Currently this is the library controller's delegate.

At the same time, many parts of the framework expect there to be only one shared library controller per media type. Thus there also is only one delegate.

In a document based application, one may thus have several object view controllers backed by only a single shared library controller. The current implementation requires a single shared delegate to provide badges for all object views. It would be much more convenient if each document could be the delegate to only the object view controllers it owns.

My current workaround is to subclass IMBImageViewController:
- (id <IMBObjectViewControllerDelegate>)delegate
  {
  if (_delegate != nil) {
      return _delegate;
  }
  
  return self.libraryController.delegate;
  }
- (void)setDelegate:(id<IMBObjectViewControllerDelegate>)delegate;
  {
  _delegate = delegate;
  }

**Comment by mikeabdullah** [comment]
I agree that the present situation isn't good.

An alternative to consider: ditch the notion of IMBObjectViewControllerDelegate. Instead, the library controller's delegate would receive messages like:

-libraryController:objectViewController:shouldDoFoo:

Kinda like how a WebView funnels delegate messages for all frames

**Comment by jjac** [comment]
Indeed, if a specific document could be an IMBObjectViewController's delegate one could take advantage of the context of that document in its delegate methods.

Regarding badges we could then facilitate information to our user like "I am using this media object in this document" instead of "I am using this media object in one of the documents currently open".
